src/System.Management.Automation/help/MUIFileSearcher.cs

```diff
@@ -113,25 +114,42 @@ private void SearchForFiles()
             }
         }
 
-        private string[] GetFiles(string path, string pattern)
-        {
+		private string[] GetFiles(string path, string pattern)
+		{ 
 #if UNIX
             // On Linux, file names are case sensitive, so we need to add
             // extra logic to select the files that match the given pattern.
             ArrayList result = new ArrayList();
             string[] files = Directory.GetFiles(path);
+
+            string regexPattern = pattern.Replace(".", @"\.");
+            regexPattern = regexPattern.Replace("*", ".*");
+            regexPattern = regexPattern.Replace("?", ".?");
+            Regex r = new Regex(regexPattern);
```

Comment:
Since this matching is infrequent, you should use an interpreted regex instead of a compiled one

https://msdn.microsoft.com/en-us/library/gg578045(v=vs.110).aspx